rocprim/CMakeLists.txt

```diff
@@ -42,12 +42,20 @@ target_include_directories(rocprim
 # This target allows using only HC interface, links only
 # against HC/HSA library, doesn't require HIP
 add_library(rocprim_hc INTERFACE)
+if(HIP_COMPILER STREQUAL "HCC")
 target_link_libraries(rocprim_hc
   INTERFACE
     rocprim
     hcc::hccrt
     hcc::hc_am
 )
+elseif(HIP_COMPILER STREQUAL "clang")
```

Comment:
Does it mean that clang also supports HC C++ API?

Reply:
HIP-Clang shouldn't support HC C++ API. Does that mean I should remove rocprim_hc entirely for HIP-Clang?

Reply:
Yes. I can check what's the best way to do that.

Reply:
I think we can safely disable it for for HIP-Clang. You disabled HC tests/benchmarks/examples anyway, I think the only other place `rocprim_hc` target is used is in `rocm_install_targets`, so add it there only if HCC is used as compiler and that's it.